fe2-android/app/src/main/java/com/github/dedis/student20_pop/model/Wallet.java

```diff
@@ -223,24 +248,29 @@ private String convertStringToPath(String string){
    *
    * @return an array of words: mnemonic sentence representing the seed for the wallet.
    */
-  public String[] exportSeed(){
-
-    SecureRandom random = new SecureRandom();
-    byte[] entropy = random.generateSeed(Words.TWELVE.byteLength());
+  public String[] exportSeed() throws GeneralSecurityException {
+    if(keysetManager != null) {
+      SecureRandom random = new SecureRandom();
+      byte[] entropy = random.generateSeed(Words.TWELVE.byteLength());
 
-    StringBuilder sb = new StringBuilder();
-    MnemonicGenerator generator = new MnemonicGenerator(English.INSTANCE);
-    generator.createMnemonic(entropy, sb::append);
+      StringBuilder sb = new StringBuilder();
+      MnemonicGenerator generator = new MnemonicGenerator(English.INSTANCE);
+      generator.createMnemonic(entropy, sb::append);
 
-    String[] words = sb.toString().split(" ");
-    Log.d(TAG,"the array of word generated:" + Arrays.toString(words));
+      String[] words = sb.toString().split(" ");
+      Log.d(TAG, "the array of word generated:" + Arrays.toString(words));
 
-    StringJoiner joiner = new StringJoiner(" ");
-    for(String i: words) joiner.add(i);
-    seed = new SeedCalculator().calculateSeed(joiner.toString(), "");
-    Log.d(TAG, "ExportSeed: new seed initialized: " + Utils.bytesToHex(seed));
+      StringJoiner joiner = new StringJoiner(" ");
+      for (String i : words)
+        joiner.add(i);
+      seed = aead.encrypt(new SeedCalculator().calculateSeed(joiner.toString(), ""),
+          new byte[0]);
+      Log.d(TAG, "ExportSeed: new seed initialized: " + Utils.bytesToHex(seed));
 
-    return words;
+      return words;
+    } else {
+      throw new IllegalArgumentException("key set manager not init!");
```

Comment:
I don't think IllegalArgumentException is the right exception, if you need an exception at all

Reply:
Did you think that instead of throw an error I can make a log.d() ?

Reply:
Sure, and return null like in `importSeed`